tests/api/test_account.py

```diff
@@ -143,6 +143,47 @@ def test_query_user(staff_api_client, customer_user, permission_manage_users):
     assert address['phone'] == user_address.phone.as_e164
 
 
+def test_non_staff_user_can_only_see_his_user_data(user_api_client,
+                                                   staff_user):
+    query = """
+    query User($id: ID!) {
+        user(id: $id) {
+            email
+        }
+    }
+    """
+    id = graphene.Node.to_global_id('User', staff_user.id)
+    variables = {'id': id}
+    response = user_api_client.post_graphql(query, variables)
+    content = get_graphql_content(response)
+    data = content['data']['user']
+    assert data is None
+
+    # Ensure user can see own data
```

Comment:
I have mixed feeling about using `ensure` in this context, at this point, you can only check if the user can see his data.

Reply:
I'm also missing an explanation for the first case, maybe we should split this test in half?